### PR TITLE
Disable publish action in forked repository

### PR DESCRIPTION
**Pull request by Goooler**
I enabled the github actions in my forked repo for build and test, but triggered publish action after jvm job, it's unnecessary to publish, so I think we can disable it in forked repo.

image

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Build
 
-on: [push, pull_request]
+on: [ push, pull_request ]
```
Comment (ZacSweers) [inline .github/workflows/build.yml:3]: Let's avoid unnecessary changes

```suggestion
on: [push, pull_request]
```

**Review by ZacSweers — APPROVED** [review]

**Review by Egorand — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/build.yml]: Reverted.
